core/tests/webdriverio_utils/waitFor.js

```diff
@@ -66,25 +64,23 @@ var elementToBeClickable = async function(element, errorMessage) {
  * @param {string} errorMessage - Error message when element is still visible.
  */
 var invisibilityOf = async function(element, errorMessage) {
-  await browser.waitUntil(
-    await until.invisibilityOf(element),
-    {
-      timeout: DEFAULT_WAIT_TIME_MSECS,
-      timeoutMsg: errorMessage
-    });
+  await element.waitForDisplayed({
+    timeout: DEFAULT_WAIT_TIME_MSECS,
+    reverse: true,
+    timeoutMsg: errorMessage
+  });
 };
 
 /**
  * Consider adding this method after each browser.url() call.
  */
 var pageToFullyLoad = async function() {
-  var loadingMessage = $('.e2e-test-loading-fullpage');
-  await browser.waitUntil(
-    await until.invisibilityOf(loadingMessage),
-    {
-      timeout: 15000,
-      timeoutMsg: 'Page takes more than 15 secs to load'
-    });
+  var loadingMessage = await $('.e2e-test-loading-fullpage');
+  await loadingMessage.waitForExist({
+    timeout: 15000,
+    reverse: true,
+    timeoutMsg: 'Pages takes more than 15 sec to load'
+  });
```

Comment:
Just confirming that we are not currently debugging this flake (we have many others to work on).

Reply:
Oops!! Sorry, I heard from @ShivamJhaa  a hour back that Chris and you are also debugging this flake. That's why I pinged both of you.

Reply:
waitForExists  returns true if the selector matches at least one element that exists in the DOM, otherwise throws an error. If the reverse flag is true, the command will instead return true if the selector does not match any elements. So here it is working as an invisivibilityOf function. PTAL [here](https://webdriver.io/docs/api/element/waitForExist)

Reply:
Okay!! Thanks!! But I still think the underlying problem lies in` var loadingMessage = $('.e2e-test-loading-fullpage');`. The moment` waitFor.pageToLoadFully` is called findElements start getting polled for this element `$('.e2e-test-loading-fullpage');`. But as default timeout as pointed by me is 0, thus the promise doesn't get resolved as no element is found. Also, if you carefully see the [error log](https://github.com/oppia/oppia/runs/7379877212?check_suite_focus=true#step:13:5067), right after the `var loadingMessage = $('.e2e-test-loading-fullpage');` there is no execution of invisbityof command and the execution directly comes out of the pageToLoadfully which makes me more confident that the problem lies in ` var loadingMessage = $('.e2e-test-loading-fullpage');`.
`
It's different from the other cases because in other cases the selectors are mostly defined at the top and then used directly inside waitForExsits and thus findElements are being polled inside waitForExisit in this case and not directly in $ selectors which are defined at the top. Can you also please confirm this? We need the terminologies in order to avoid the future flakes of this kind too.

Reply:
I also thinks the same, but can we do this debugging in some other PR, also the changes done for removing the flake is helping the other test to pass like blogDashboard so I think we should keep the changes.
Thanks

Reply:
I agree that we should do the debugging in some other PR now because Shivam has other PR's stacked on this and also keeping waitForExists since visibiltyof in webdriverio causes flakiness.

Reply:
Thanks

Reply:
Also, Do we not have reverse in waitForDisplayed because this seems to be the alternative for invisibilityof?

Reply:
Yes, we have. And its already added.